test/acceptance/integration/steps_lib/general.py

```diff
@@ -0,0 +1,93 @@
+# -*- coding: utf-8 -*-
+"""
+Copyright 2014 Telefonica Investigación y Desarrollo, S.A.U
+
+This file is part of fiware-orion-pep
+
+fiware-orion-pep is free software: you can redistribute it and/or
+modify it under the terms of the GNU Affero General Public License as
+published by the Free Software Foundation, either version 3 of the License,
+or (at your option) any later version.
+
+fiware-orion-pep is distributed in the hope that it will be useful,
+but WITHOUT ANY WARRANTY; without even the implied warranty of
+MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+See the GNU Affero General Public License for more details.
+
+You should have received a copy of the GNU Affero General Public
+License along with fiware-orion-pep.
+If not, seehttp://www.gnu.org/licenses/.
+
+For those usages not covered by the GNU Affero General Public License
+please contact with::[iot_support@tid.es]
+"""
+from iotqautils.cb_utils import CbNgsi10Utils, CbNgsi9Utils
+import requests
+from integration.tools.general_utils import check_key_value, drop_database
+
+__author__ = 'Jon'
+
+from lettuce import step, world
+
+
+@step('a new NGSI version "([^"]*)" petition with the service "([^"]*)" and the subservice "([^"]*)"')
+def a_new_ngsi10_petition_with_the_service_and_the_subservice(step, ngsi_version, service, subservice):
+    world.entities = None
+    world.attributes_consult = None
+    world.attributes_creation = None
+    world.context_elements = None
+    world.notify_conditions = None
+    world.context_registration = None
+    world.mock_data = None
+    if service == 'empty':
+        world.service = None
+    else:
+        world.service = service
+    if subservice == 'empty':
+        world.subservice = None
+    else:
+        world.subservice = subservice
+    world.payloads_count += 1
+    world.response_count += 1
+    world.cb_count += 1
+    if ngsi_version == '10':
+        world.cb[world.cb_count] = CbNgsi10Utils(world.config['context_broker']['host'], world.service,
+                                                 world.subservice,
+                                                 port=world.config['context_broker']['port'])
+    elif ngsi_version == '9':
+        world.cb[world.cb_count] = CbNgsi9Utils(world.config['context_broker']['host'], world.service, world.subservice,
+                                                port=world.config['context_broker']['port'])
+    else:
+        raise ValueError(
+            'The version of ngsi api have to be \'9\' or \'10\', not {ngsi_version}'.format(ngsi_version=ngsi_version))
```

Comment:
So maybe a rewording could be:

"NGSI sub-api number have to be \'9\' or \'10\', not"

What I don'tl like is to use "version" there... it could be confused with the "v1" (the actual API version) in the URL, for example.